app/src/main/java/org/thoughtcrime/securesms/megaphone/Megaphones.java

```diff
@@ -78,11 +79,13 @@ private Megaphones() {}
    * This is when you would hide certain megaphones based on {@link FeatureFlags}. You could
    * conditionally set a {@link ForeverSchedule} set to false for disabled features.
    */
-  private static Map<Event, MegaphoneSchedule> buildDisplayOrder() {
+  private static Map<Event, MegaphoneSchedule> buildDisplayOrder(Context context) {
     return new LinkedHashMap<Event, MegaphoneSchedule>() {{
       put(Event.REACTIONS, new ForeverSchedule(true));
       put(Event.PINS_FOR_ALL, new PinsForAllSchedule());
-      put(Event.PIN_REMINDER, new SignalPinReminderSchedule());
+      if (!TextSecurePreferences.isPinV2ReminderDisabled(context)) {
+        put(Event.PIN_REMINDER, new SignalPinReminderSchedule());
+      }
```

Comment:
An easy way to fix the problem you mentioned in the PR is to modify the class `SignalPinReminderSchedule` instead. Add something like this to the method `shouldDisplay`:
```
if (TextSecurePreferences.isPinV2ReminderDisabled(ApplicationDependencies.getApplication())) {
  return false;
}
```

Reply:
I didn't see this way to get the context to this class myself, I'll test this approach.